src/data/mod.rs

```diff
@@ -55,11 +59,66 @@ pub trait Subscriber {
     async fn exit(self) -> Result<(), AstarteError>;
 }
 
+/// Store errors
+#[derive(Debug, thiserror::Error, displaydoc::Display)]
+pub enum StoreError {
+    /// couldn't connect to Sqlite store
+    Connect(#[source] SqliteError),
+    /// Path is not UTF-8, `{0}`
+    PathUtf8(PathBuf),
+}
+
+/// Options to connect a store to the Astarte device sdk or to the Message Hub nodes
+pub struct ConnectOptions<'a> {
+    interface_dir: &'a Path,
+    store_dir: &'a Path,
+}
+
+impl<'a> ConnectOptions<'a> {
+    pub fn new<I, S>(interface_dir: &'a I, store_dir: &'a S) -> Self
+    where
+        I: AsRef<Path>,
+        S: AsRef<Path>,
+    {
+        Self {
+            interface_dir: interface_dir.as_ref(),
+            store_dir: store_dir.as_ref(),
+        }
+    }
+}
```

Comment:
is it really useful? the store_directory doesn't appear used in other parts, so is it necessary to store it into ConnectionOptions? I think also you can remove this struct and pass only interface_directory directly to connect method.

Reply:
No, it is not necessary. It was meant to put together all the information related to the establishment of a connection